packages/cli/src/utils/config.js

```diff
@@ -0,0 +1,97 @@
+import path from 'path'
+import consola from 'consola'
+import defaultsDeep from 'lodash/defaultsDeep'
+import { defaultNuxtConfigFile, getDefaultNuxtConfig } from '@nuxt/config'
+import esm from 'esm'
+
+export async function loadNuxtConfig(argv) {
+  const rootDir = path.resolve(argv._[0] || '.')
+  let nuxtConfigFile
+  let options = {}
+
+  try {
+    nuxtConfigFile = require.resolve(path.resolve(rootDir, argv['config-file']))
+  } catch (e) {
+    if (e.code !== 'MODULE_NOT_FOUND') {
+      throw (e)
+    } else if (argv['config-file'] !== defaultNuxtConfigFile) {
+      consola.fatal('Could not load config file: ' + argv['config-file'])
+    }
+  }
+
+  if (nuxtConfigFile) {
+    if (nuxtConfigFile.endsWith('.ts')) {
+      options = require(nuxtConfigFile) || {}
+    } else {
+      clearRequireCache(nuxtConfigFile)
+      options = esm(module, { cache: false, cjs: { cache: false } })(nuxtConfigFile) || {}
+    }
+
+    if (options.default) {
+      options = options.default
+    }
+
+    if (typeof options === 'function') {
+      try {
+        options = await options()
+        if (options.default) {
+          options = options.default
+        }
+      } catch (error) {
+        consola.error(error)
+        consola.fatal('Error while fetching async configuration')
+      }
+    }
+
+    // Keep _nuxtConfigFile for watching
+    options._nuxtConfigFile = nuxtConfigFile
+
+    // Keep all related files for watching
+    options._nuxtConfigFiles = Array.from(scanRequireTree(nuxtConfigFile))
+  }
+
+  if (typeof options.rootDir !== 'string') {
+    options.rootDir = rootDir
+  }
+
+  // Nuxt Mode
+  options.mode =
+    (argv.spa && 'spa') || (argv.universal && 'universal') || options.mode
+
+  // Server options
+  options.server = defaultsDeep({
+    port: argv.port || undefined,
+    host: argv.hostname || undefined,
+    socket: argv['unix-socket'] || undefined
+  }, options.server || {}, getDefaultNuxtConfig().server)
+
+  return options
+}
+
+function clearRequireCache(id) {
+  const entry = require.cache[id]
+  if (!entry || id.includes('node_modules')) {
+    return
+  }
+
+  for (const child of entry.children) {
+    clearRequireCache(child.id)
+  }
+
+  delete require.cache[id]
```

Comment:
@pi0 
How do you think delete `nuxt.config.js` module from its parent ?

For example:
```js
const parent = require.cache[nuxtConfigFile].parent
delete require.cache[nuxtConfigFile]
console.log(parent.children) // `nuxt.config.js` module still exists in `children`
```

Reply:
It is an unsafe operation to modify parent. Clearing from global cache just works fine :)

Reply:
I'm just worrying about the old config module is still in memory and maybe can not be collected.

Reply:
You are right. Despite [docs](https://nodejs.org/api/modules.html#modules_require_cache) parent keeps a reference of all old child. It is fixed now.